Assets/MixedRealityToolkit/Definitions/SpatialAwareness/MixedRealitySpatialAwarenessMeshObserverProfile.cs

```diff
@@ -0,0 +1,136 @@
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Attributes;
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Definitions.SpatialAwarenessSystem
+{
+    /// <summary>
+    /// Configuration profile settings for spatial awareness mesh observers.
+    /// </summary>
+    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Mixed Reality Spatial Awareness Mesh Observer Profile", fileName = "MixedRealitySpatialAwarenessMeshObserverProfile", order = (int)CreateProfileMenuItemIndices.SpatialAwarenessMeshObserver)]
+    public class MixedRealitySpatialAwarenessMeshObserverProfile : BaseMixedRealityProfile 
+    {
+        #region IMixedRealitySpatialAwarenessObserver settings
+
+        [SerializeField]
+        [Tooltip("How should the observer behave at startup?")]
+        private AutoStartBehavior startupBehavior = AutoStartBehavior.AutoStart;
+
+        /// <summary>
+        /// Indicates if the observer is to start immediately or wait for manual startup.
+        /// </summary>
+        public AutoStartBehavior StartupBehavior => startupBehavior;
+
+        [SerializeField]
+        [Tooltip("Should the spatial observer remain in a fixed location?")]
+        private bool isStationaryObserver = false;
+
+        /// <summary>
+        /// Indicates whether or not the spatial observer is to remain in a fixed location.
+        /// </summary>
+        public bool IsStationaryObserver => isStationaryObserver;
+
+        [SerializeField]
+        [Tooltip("The dimensions of the spatial observer volume, in meters.")]
+        private Vector3 observationExtents = Vector3.one * 3;
+
+        /// <summary>
+        /// The size of the volume, in meters per axis, from which individual observations will be made.
+        /// </summary>
+        public Vector3 ObservationExtents => observationExtents;
+
+        [SerializeField]
+        [Tooltip("The shape of observation volume")]
+        private VolumeType observerVolumeType = VolumeType.AxisAlignedCube;
+
+        /// <summary>
+        /// The shape (ex: axis aligned cube) of the observation volume.
+        /// </summary>
+        public VolumeType ObserverVolumeType => observerVolumeType;
+
+
+        [SerializeField]
+        [Tooltip("How often, in seconds, should the spatial observer update?")]
+        private float updateInterval = 3.5f;
+
+        /// <summary>
+        /// The frequency, in seconds, at which the spatial observer updates.
+        /// </summary>
+        public float UpdateInterval => updateInterval;
+
+        #endregion IMixedRealitySpatialAwarenessObserver settings
+
+        #region IMixedRealitySpatialAwarenessMeshObserver settings
+
+        [PhysicsLayer]
+        [SerializeField]
+        [Tooltip("Physics layer on which to set observed meshes.")]
+        private int meshPhysicsLayer = 31;
```

Comment:
Nit: this is probably not a const expression, but would be a bit more consumable: https://docs.unity3d.com/ScriptReference/LayerMask.NameToLayer.html

Reply:
besides not being a const. the name might be changed by the user....